pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -0,0 +1,47 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package realizestate
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+type RealizeStateService struct {
+	common.Service
+}
+
+func InitializeRealizeState(service common.Service) *RealizeStateService {
+	return &RealizeStateService{
+		Service: service,
+	}
+}
+
+// CheckRealizeState allows the caller to check realize status of entityType with retries.
+// backoff defines the maximum retries and the wait interval between two retries.
+func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, orgID, projectID, intentPath, entityType string) error {
+	return retry.OnError(backoff, func(err error) bool {
+		return err != nil
+	}, func() error {
+		results, err := service.NSXClient.RealizedStateClient.List(orgID, projectID, intentPath, nil)
```

Comment:
In my code, I imported `"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/realized_state"`, so the `List` function only needs the path, e.g.
`realizedPort, err := service.NSXClient.RealizedEntitiesClient.List(*nsxSubnetPort.Path, nil)`
Which one do you prefer to?

Reply:
I didn't try `infra/realize_state`, but I think for VPC, we should always use the client under `orgs/projects/infra`?

Reply:
It's ok for me.